File: ruby/practice/classic_puzzle/easy/onboarding.rb
STDOUT.sync = true # DO NOT REMOVE

# game loop
loop do
  enemy_1 = gets.chomp # name of enemy 1
  dist_1  = gets.to_i  # distance to enemy 1
  enemy_2 = gets.chomp # name of enemy 2
  dist_2  = gets.to_i  # distance to enemy 2

  # Write an action using puts
  if dist_1 < dist_2
    puts enemy_1
  else
    puts enemy_2
  end

  # Enter the code here
end

File: perl/practice/classic_puzzle/easy/onboarding.pl
select(STDOUT); $| = 1; # DO NOT REMOVE

# game loop
while (1) {
    chomp(my $enemy_1 = <STDIN>); # name of enemy 1
    chomp(my $dist_1 = <STDIN>); # distance to enemy 1
    chomp(my $enemy_2 = <STDIN>); # name of enemy 2
    chomp(my $dist_2 = <STDIN>); # distance to enemy 2

    # Write an action using print
    # To debug: print STDERR "Debug messages...\n";

    # Enter the code here

    if ($dist_1 < $dist_2) {
        print "$enemy_1\n";
    } else {
        print "$enemy_2\n";
    }
}
